packages/gatsby/src/redux/actions/public.js

```diff
@@ -423,14 +423,17 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
   const contextModified =
     !!oldPage && !_.isEqual(oldPage.context, internalPage.context)
 
-  const alternateSlashPath = page.path.endsWith(`/`)
-    ? page.path.slice(0, -1)
-    : page.path + `/`
-
-  if (store.getState().pages.has(alternateSlashPath)) {
+  // This should only check against existing pages with
+  // trailing slashes. Creating an alternate page with
+  // a trailing slash can often be fine because
+  // it will get replaced later in the process
+  // by gatsby-plugin-trailing-slashes.
+  if (!page.path.endsWith(`/`) && store.getState().pages.has(page.path + `/`)) {
```

Comment:
You kinda bypass the check now, right? If I create a duplicate of a page I would not see an error anymore which is the reason why we added this warning to make people aware that you could have two pages at the same route.